linera-execution/src/wasm/runtime_actor/handlers.rs

```diff
@@ -66,7 +70,7 @@ where
 }
 
 #[async_trait]
-impl<Runtime> RequestHandler<ContractRequest> for &Runtime
+impl<Runtime> RequestHandler<ContractRequest> for RwLock<&Runtime>
```

Comment:
`RwLock<&Runtime>` is a little suspicious as type. Maybe it should be `RwLock<&mut Runtime>` or `RwLock<Runtime>`?

On the same note, there is a lot interior mutability in `struct ExecutionRuntime` but when I remove `derive(Clone)` it appears to still compile (before this PR).